tests/python/unittest/test_io.py

```diff
@@ -306,12 +306,13 @@ def check_CSVIter_synthetic():
                 fout.write('0\n')
 
         data_train = mx.io.CSVIter(data_csv=data_path, data_shape=(8,8),
-                                   label_csv=label_path, batch_size=100)
-        expected = mx.nd.ones((100, 8, 8))
+                                   label_csv=label_path, batch_size=100, dtype=dtype)
+        expected = mx.nd.ones((100, 8, 8), dtype=dtype)
         for batch in iter(data_train):
             assert_almost_equal(data_train.getdata().asnumpy(), expected.asnumpy())
```

Comment:
we may need to set some atol and rtol

Reply:
We need the exact result from parsing, the fundamental cause for this PR was that using float32 for parsing large integers is inaccurate.

Reply:
Does this test actually check the dtype? assert_almost_equal only checks the value but not the dtype
Suggest test:
1. assert(data_train.getdata() == dtype)
2. When dtype is int32, generate large int like 50000001 and test if the data loaded is exactly 50000001.

Reply:
Even if you don't implement int32 parser this test will pass:
```
>>> a = mx.nd.ones((2,2), dtype='int32')
>>> b = mx.nd.ones((2,2), dtype='float32')
>>> mx.test_utils.assert_almost_equal(a.asnumpy(), b.asnumpy())
>>> # no exception
```